go/exocom/src/security/security_test.go

```diff
@@ -2,13 +2,44 @@ package security_test
 
 import (
 	"github.com/Originate/exocom/go/exocom/src/security"
+	"github.com/Originate/exocom/go/structs"
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 )
 
 var _ = Describe("Security", func() {
-	It("creates a security manager", func() {
-		securityManager := security.NewSecurityManager()
-		Expect(securityManager).ToNot(BeNil())
+	Describe("ReceiveMessage", func() {
+		var (
+			manager         *security.Manager
+			incomingMessage structs.Message
+			securityRequest *security.Result
+			securityResult  *security.Result
+		)
+
+		BeforeEach(func() {
+			manager = security.NewSecurityManager()
+			incomingMessage = structs.Message{
+				Name: "name",
+			}
+			securityRequest = manager.ReceiveMessage(incomingMessage)
+		})
+
+		It("finds and returns authorized messages", func() {
+			message := structs.Message{
+				Name:       "message authorized",
+				ActivityID: securityRequest.MessageToSend.ActivityID,
+			}
+			securityResult = manager.ReceiveMessage(message)
+			Expect(securityResult.MessageToSend).To(Equal(incomingMessage))
+			Expect(securityResult.WarningMessage).To(Equal(""))
+		})
+		It("finds and returns authorized messages", func() {
+			message := structs.Message{
+				Name: "message unauthorized",
+			}
+			securityResult = manager.ReceiveMessage(message)
+			Expect(securityResult.MessageToSend).To(Equal(structs.Message{}))
+			Expect(securityResult.WarningMessage).To(Equal("Warning: Unauthorized message '' from '' with activityId ''"))
```

Comment:
Can we update this test case so the warning message is entirely filled out? I think we need to pass an activityId with "message unauthorized" and set a Sender on the initialMessage